File: Advanced Methods of Programming/Toy Language Interpreter/src/Model/Statements/Assignment.java
package Model.Statements;

import Model.Expressions.IExpression;
import Model.ProgramState;

public class Assignment implements IStatement {
    private String var;
    private IExpression exp;

    public Assignment(String var, IExpression exp){
        this.var = var;
        this.exp = exp;
    }

    @Override
    public String toString() {
        return var + "=" + exp.toString();
    }

    @Override
    public ProgramState exec(ProgramState state) {
        int value = exp.eval(state);
        state.getSymTable().put(var, value);
        return state;
    }
}

File: Advanced Methods of Programming/CSharp Toy Language Interpreter/ToyLanguageInterpreter/Src/Model/Statements/Assignment.cs
using ToyLanguageInterpreter.Src.Model.Expressions;

namespace ToyLanguageInterpreter.Src.Model.Statements
{
    class Assignment : IStatement
    {
        private string var;
        private IExpression exp;

        public Assignment(string var, IExpression exp)
        {
            this.var = var;
            this.exp = exp;
        }

        public ProgramState Exec(ProgramState state)
        {
            int value = exp.Eval(state);
            state.SymTable.Put(var, value);
            return null;
        }

        public override string ToString()
        {
            return var + "=" + exp.ToString(); 
        }
    }
}
